emcc.py

```diff
@@ -1217,7 +1201,28 @@ def check(input_file):
           newargs.append('-disable-llvm-optzns')
 
       if not shared.Settings.LEGALIZE_JS_FFI:
-        assert shared.Building.is_wasm_only(), 'LEGALIZE_JS_FFI incompatible with RUNNING_JS_OPTS and non-wasm BINARYEN_METHOD.'
+        assert shared.Settings.WASM_ONLY, 'LEGALIZE_JS_FFI incompatible with RUNNING_JS_OPTS and non-wasm BINARYEN_METHOD.'
+
+      if shared.Settings.BINARYEN:
+        # determine wasm-only mode, now that all settings are settled on
+        shared.Settings.WASM_ONLY = shared.Building.is_wasm_only()
```

Comment:
emcc.py order was a little wrong, thanks, fixed.

And clarified in settings.js what is internal use only.